xmppserver/src/main/java/org/dom4j/io/XMPPPacketReader.java

```diff
@@ -77,6 +80,30 @@ public XMPPPacketReader(DocumentFactory factory) {
         this.factory = factory;
     }
 
+    /**
+     * Retrieves a collection of namespaces declared in the current element that the parser is on, that are not defined
+     * in {@link #IGNORED_NAMESPACE_ON_STANZA}
+     *
+     * @param xpp the parser
+     * @return A collection of namespaces
+     * @throws XmlPullParserException
+     */
+    @Nonnull
+    public static Set<Namespace> getNamespacesOnCurrentElement(XmlPullParser xpp) throws XmlPullParserException
```

Comment:
I reckon this method would be a great place for a unit test. I can see what it should be doing, but a unit test would at least validate it does what I think it does.

Reply:
Agreed! I've added some in a new commit. If anything, it showed that I could have better named the method (which I now did).